src/opm/parser/eclipse/EclipseState/Schedule/RFTConfig.cpp

```diff
@@ -0,0 +1,176 @@
+
+  OPM is distributed in the hope that it will be useful,
+
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/parser/eclipse/EclipseState/Schedule/TimeMap.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/RFTConfig.hpp>
+
+namespace Opm {
+
+RFTConfig::RFTConfig(const TimeMap& time_map) :
+    tm(time_map)
+{
+}
+
+bool RFTConfig::rft(const std::string& well_name, std::size_t report_step) const {
+    if (report_step >= this->tm.size())
+        throw std::invalid_argument("Invalid ");
+
+    const auto well_iter = this->well_open.find(well_name);
+    if (well_iter != this->well_open.end()) {
+
+        // A general "Output RFT when the well is opened" has been configured with WRFT
+        if (this->well_open_rft_time.first && this->well_open_rft_time.second <= report_step) {
+            if (well_iter->second == report_step)
+                return true;
+        }
+
+
+        // A FOPN setting has been configured with the WRFTPLT keyword
+        if (this->well_open_rft_name.count(well_name) > 0) {
+            if (well_iter->second == report_step)
+                return true;
+        }
+    }
+
+    if (this->rft_config.count(well_name) == 0)
+        return false;
+
+    auto rft_pair = this->rft_config.at(well_name)[report_step];
+    if (rft_pair.first == RFTConnections::YES)
+        return (rft_pair.second == report_step);
+
+    if (rft_pair.first == RFTConnections::NO)
+        return false;
+
+    if (rft_pair.first == RFTConnections::REPT)
+        return true;
+
+    if (rft_pair.first == RFTConnections::TIMESTEP)
+        return true;
+
+    return false;
+}
+
+bool RFTConfig::plt(const std::string& well_name, std::size_t report_step) const {
+    if (report_step >= this->tm.size())
+        throw std::invalid_argument("Invalid ");
+
+    if (this->plt_config.count(well_name) == 0)
+        return false;
+
+    auto plt_pair = this->plt_config.at(well_name)[report_step];
+    if (plt_pair.first == PLTConnections::YES)
+        return (plt_pair.second == report_step);
+
+    if (plt_pair.first == PLTConnections::NO)
+        return false;
+
+    if (plt_pair.first == PLTConnections::REPT)
+        return true;
+
+    if (plt_pair.first == PLTConnections::TIMESTEP)
+        return true;
+
+    return false;
+}
+
+void RFTConfig::updateRFT(const std::string& well_name, std::size_t report_step, RFTConnections::RFTEnum value) {
+    if (value == RFTConnections::FOPN)
+        this->setWellOpenRFT(well_name);
+    else {
+        if (this->rft_config.count(well_name) == 0) {
+            auto state = DynamicState<std::pair<RFTConnections::RFTEnum, std::size_t>>(this->tm, std::make_pair(RFTConnections::NO, 0));
+            this->rft_config.emplace( std::make_pair(well_name, state) );
```

Comment:
You probably don't need `emplace(make_pair(args))` here&mdash;`emplace(args)` ought to be sufficient since `RFTConfig::rft_config` is an `unordered_map`